### ISSUE TITLE
golangci-lint reports nothing when prerequisities aren't met

### ISSUE THREAD

**Issue by jm33-m0** [golangci/golangci-lint#867]
Code:
```go
package main

import "fmt"

func yes() bool {
	return 1+1 > 2 || 1+1 == 2
}

func main() {
	if yes() {
		fmt.Println("yes")
		foo // undeclared name
	}
}
```
golangci-lint has **too many prerequisities**, why can't it just report `undeclared name` to me instead?

Reporting with a `WARN` makes it impossible for tools like [ALE](https://github.com/dense-analysis/ale) to parse, as a result, users see **no error reports at all**, which is misleading.

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```bash
$ golangci-lint --version
golangci-lint has version v1.21.1-0.20191111000716-14ebae29060f built from (unknown, mod sum: "h1:J9Xwp1uY0EooAwj8OONV3bbgfAChsFdOMfFzH5eWcnQ=") on (unknown)
```

</details>

<details><summary>Config file</summary>

```bash
$ cat .golangci.yml
```

</details>

<details><summary>Go environment</summary>

```bash
$ go version && go env

```

</details>

<details><summary>Verbose output of running</summary>

```bash
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /tmp /] 
INFO [lintersdb] Active 10 linters: [deadcode errcheck gosimple govet ineffassign staticcheck structcheck typecheck unused varcheck] 
INFO [lintersdb] Active 10 linters: [deadcode errcheck gosimple govet ineffassign staticcheck structcheck typecheck unused varcheck] 
INFO [loader] Go packages loading at mode 575 (compiled_files|deps|files|imports|name|types_sizes|exports_file) took 56.242358ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 88.759µs 
INFO [runner/goanalysis_metalinter/goanalysis] analyzers took 1.820877ms with top 10 stages: fact_deprecated: 262.478µs, tokenfileanalyzer: 48.736µs, buildssa: 34.165µs, printf: 33.154µs, SA9002: 29.204µs, SA1015: 28.375µs, SA4004: 27.553µs, SA4021: 25.55µs, SA1025: 25.341µs, S1016: 25.17µs 
WARN [runner] Can't run linter goanalysis_metalinter: SA1013: failed prerequisites: inspect@_/tmp 
INFO [runner/unused/goanalysis] analyzers took 21.593µs with top 10 stages: U1000: 10.888µs, buildssa: 10.705µs 
WARN [runner] Can't run linter unused: buildssa: analysis skipped: errors in package: [/tmp/g.go:12:3: undeclared name: foo] 
INFO [runner] processing took 2.841µs with stages: max_same_issues: 546ns, diff: 345ns, skip_dirs: 244ns, cgo: 227ns, filename_unadjuster: 225ns, max_from_linter: 215ns, nolint: 182ns, identifier_marker: 144ns, path_prettifier: 135ns, skip_files: 135ns, autogenerated_exclude: 133ns, exclude: 57ns, uniq_by_line: 56ns, max_per_file_from_linter: 51ns, path_shortener: 49ns, source_code: 49ns, exclude-rules: 48ns 
INFO [runner] linters took 97.609384ms with stages: goanalysis_metalinter: 61.670538ms, unused: 35.901117ms 
INFO File cache stats: 0 entries of total size 0B 
INFO Memory: 3 samples, avg is 69.4MB, max is 69.7MB 
INFO Execution took 160.230217ms   
```

</details>

**Comment by ernado** [comment]
This should error in latest versions, we've switched from warnings to errors in this place.
